Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -331,6 +345,148 @@ public LibraryDependenciesResult(@Nullable final String[] libraries,
     }
   }
 
+  private void configureImportedLibraries(@NotNull final Collection<String> rootsToAddToLib) {
+    runInEventDispatchThread(new Runnable() {
+      @Override
+      public void run() {
+        ApplicationManager.getApplication().runWriteAction(new Runnable() {
+          public void run() {
+            doConfigureImportedLibraries(rootsToAddToLib);
+          }
+        });
+      }
+    });
+  }
+
+  private static void runInEventDispatchThread(final Runnable runnable) {
+    try {
+      if (SwingUtilities.isEventDispatchThread()) {
+        runnable.run();
+      }
+      else {
+        ApplicationManager.getApplication().invokeAndWait(new Runnable() {
+          public void run() {
+            runnable.run();
+          }
+        }, ModalityState.defaultModalityState());
+      }
+    }
+    catch (Exception e) {
+      LOG.warn(e);
+    }
+  }
+
+  private void doConfigureImportedLibraries(@NotNull final Collection<String> libraries) {
+    for (final Project project : myRootsHandler.getTrackedProjects()) {
```

Comment:
One project with a lot of modules of different types is ok. Several open projects where only one contains Dart is also ok (DartRootsHandler.getTrackedProjects() will return only one in this case). More than one open project - so far no idea how to configure correctly. I'm afraid we have to skip auto-configuration in this case for now.